nomos-libp2p/src/behaviour/nat/state_machine.rs

```diff
@@ -0,0 +1,749 @@
+    fn on_event(self: Box<Self>, event: Event, command_tx: &CommandTx) -> Box<dyn OnEvent<Event>> {
+        match event {
+            Event::ExternalAddressConfirmed(addr) if self.state.addr_to_test() == &addr => {
+                command_tx.send(Command::ScheduleAutonatClientTest(addr));
+                self.boxed(TestIfMappedPublic::into_mapped_public)
+            }
+            Event::AutonatClientTestFailed(addr) if self.state.addr_to_test() == &addr => {
+                self.boxed(TestIfMappedPublic::into_private)
+            }
+            Event::ExternalAddressConfirmed(addr) | Event::AutonatClientTestFailed(addr) => {
+                tracing::error!(
+                    "Autonat client reported address {}, but {} was expected",
+                    addr,
+                    self.state.addr_to_test(),
+                );
+                self
+            }
+            _ => self,
+        }
+    }
+}
+
+impl OnEvent<Event> for State<Public, Event> {
+    fn on_event(self: Box<Self>, event: Event, command_tx: &CommandTx) -> Box<dyn OnEvent<Event>> {
+        match event {
+            Event::ExternalAddressConfirmed(addr) | Event::AutonatClientTestOk(addr)
+                if self.state.addr() == &addr =>
+            {
+                command_tx.send(Command::ScheduleAutonatClientTest(addr));
+                self
+            }
+            Event::AutonatClientTestFailed(addr) if self.state.addr() == &addr => {
+                self.boxed(Public::into_test_if_public)
+            }
+            Event::ExternalAddressConfirmed(addr) => {
+                tracing::error!(
+                    "Swarm confirmed external address {}, but {} was expected",
+                    addr,
+                    self.state.addr(),
+                );
+                self
+            }
+            Event::AutonatClientTestOk(addr) | Event::AutonatClientTestFailed(addr) => {
+                tracing::error!(
+                    "Autonat client reported address {}, but {} was expected",
+                    addr,
+                    self.state.addr(),
+                );
+                self
+            }
+            _ => self,
+        }
+    }
+}
+
+impl OnEvent<Event> for State<MappedPublic, Event> {
+    fn on_event(self: Box<Self>, event: Event, command_tx: &CommandTx) -> Box<dyn OnEvent<Event>> {
+        match event {
+            Event::ExternalAddressConfirmed(addr) | Event::AutonatClientTestOk(addr)
+                if self.state.addr() == &addr =>
+            {
+                command_tx.send(Command::ScheduleAutonatClientTest(addr));
+                self
+            }
+            Event::AutonatClientTestFailed(addr) if self.state.addr() == &addr => {
+                self.boxed(MappedPublic::into_test_if_public)
+            }
+            Event::ExternalAddressConfirmed(addr) => {
+                tracing::error!(
+                    "Swarm confirmed external address {}, but {} was expected",
```

Comment:
Should you use different log messages for different states, so we can easily distinguish between them? Or at the very least the state we are logging for.

Reply:
Nah I think we might want to grep all logs for the state machine, so a target like `nomos-network::nat::state-machine` should be sufficient. I don' see any case in which we might want to grep logs for a specific state. If anything, the opposite so we can see what is going on. I would perhaps just add the state in there or use a different message.